### PR TITLE
fix(cli): Fix appset generate in --core mode

### LINKED ISSUE #20692: argocd appset generate doesn't work when login with --core
Checklist:

* [ ] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [ ] I've included steps to reproduce the bug.
* [ ] I've pasted the output of `argocd version`.

**Describe the bug**

The appset generate from the argocd CLI doesn't work when being logged with the `--core` option.

**To Reproduce**

- argocd login --core
- argocd appset generate anyappset.yaml

**Expected behavior**

Should generate the appset

**Screenshots**

**Version**
2.13.0

**Logs**
```
error generating params error="error getting project default: no kind is registered for the type v1alpha1.AppProject in scheme \"pkg/runtime/scheme.go:100\""
```

### PR DESCRIPTION
**Pull request by OpenGuidou**
Fixes #20692

I'd like it to be cherry-picked to 2.13.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [x] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [x] Optional. My organization is added to USERS.md.
* [x] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

**Comment by speedfl** [comment]
LGTM

File: cmd/argocd/commands/headless/headless.go [hunk 1]
```diff
@@ -237,7 +238,14 @@ func MaybeStartLocalServer(ctx context.Context, clientOpts *apiclient.ClientOpti
 		return fmt.Errorf("error creating kubernetes dynamic clientset: %w", err)
 	}
 
-	controllerClientset, err := client.New(restConfig, client.Options{})
+	scheme := runtime.NewScheme()
+	err = v1alpha1.AddToScheme(scheme)
```
Comment (andrii-korotkov-verkada) [inline cmd/argocd/commands/headless/headless.go:242]: Why do we need to do this? This seems like something that should be done automatically on some init.

Comment (OpenGuidou) [reply cmd/argocd/commands/headless/headless.go:242]: It's needed in the headless mode, when login with the --core option.
On this case you use this local api server and not the real remote one.
It's not required in the regular mode

**Review by andrii-korotkov-verkada — APPROVED** [review]

**Review by ishitasequeira — APPROVED** [review]

**Pull request merged**